src/sidebars/search/search-controller.js

```diff
@@ -19,15 +19,23 @@
             this._watch();
         }
 
+        hasStore() {
```

Comment:
Was thinking of just a method called `getFavourites` that does what your `hasStore` method did, but adds the extra line of returning the favourites? If it can't get them then can return null/undefined as per the suggestion for `getStore`